### PR TITLE
add primitive_job_id field in Job

### PR DESCRIPTION
**Pull request by akihikokuroda**
### Summary
A part of : #895 

### Details and comments
Add primitive job id field to Job model.  It keeps the list of primitive job ids in a string format.
Add "program job id" to ray runtime environment so that the primitive request can include it.

```
env = json.loads(os.environ.get("RAY_JOB_CONFIG_JSON_ENV_VAR"))
estimator.run(ansatz, hamiltonian, options={"job_id": env["runtime_env"]["quantum_middleware_job"]}, parameter_values=params).result().values[0]
```

### CODE REVIEW

File: gateway/tests/api/test_job.py [hunk 1]
```diff
@@ -88,3 +88,40 @@ def test_stop_job(self):
...
+            reverse(
+                "v1:jobs-add-primitive", args=["1a7947f9-6ae8-4e3d-ac1e-e7d608deec82"]
+            ),
+            format="json",
+            data={"primitive": "primitive-id-2"},
+        )
+        self.assertEqual(jobs_response.status_code, status.HTTP_200_OK)
+        self.assertEqual(
+            jobs_response.data.get("primitive_job_id"),
+            '"primitive-id","primitive-id-2"',
```
Comment (IceKhan13) [inline gateway/tests/api/test_job.py:116]: should we store it as json list? 
`["primitive_id_1", "primitive_id_2"]`

File: client/quantum_serverless/core/job.py [hunk 2]
```diff
@@ -96,6 +96,14 @@ def result(self, job_id: str):
         """Return results."""
         raise NotImplementedError
 
+    def primitive(self, job_id: str):
```
Comment (IceKhan13) [inline client/quantum_serverless/core/job.py:99]: Why do we need this functions?

File: gateway/tests/api/test_job.py [hunk 3]
```diff
@@ -88,3 +88,40 @@ def test_stop_job(self):
         ).first()
         self.assertEqual(job.status, Job.STOPPED)
         self.assertEqual(job_stop_response.data.get("message"), "Job has been stopped.")
+
+    def test_job_add_primitive(self):
```
Comment (IceKhan13) [inline gateway/tests/api/test_job.py:92]: ah, I love tests 😇

File: gateway/tests/api/test_job.py (hunk 1 above)
Comment (akihikokuroda) [reply gateway/tests/api/test_job.py:116]: I'll change it.

File: client/quantum_serverless/core/job.py (hunk 2 above)
Comment (akihikokuroda) [reply client/quantum_serverless/core/job.py:99]: Oh, yes, It is not necessary.  I'll take it out.

**Pull request closed without merging**